aws_lambda_powertools/utilities/parameters/secrets.py

```diff
@@ -117,6 +123,135 @@ def _get_multiple(self, path: str, **sdk_options) -> Dict[str, str]:
         """
         raise NotImplementedError()
 
+    def _create_secret(self, name: str, **sdk_options):
+        """
+        Create a secret with the given name.
+
+        Parameters:
+        ----------
+        name: str
+            The name of the secret.
+        **sdk_options:
+            Additional options to be passed to the create_secret method.
+
+        Raises:
+            SetSecretError: If there is an error setting the secret.
+        """
+        try:
+            sdk_options["Name"] = name
+            return self.client.create_secret(**sdk_options)
+        except Exception as exc:
+            raise SetSecretError(f"Error setting secret - {str(exc)}") from exc
+
+    def _update_secret(self, name: str, **sdk_options):
+        """
+        Update a secret with the given name.
+
+        Parameters:
+        ----------
+        name: str
+            The name of the secret.
+        **sdk_options:
+            Additional options to be passed to the create_secret method.
+        """
+        sdk_options["SecretId"] = name
+        return self.client.put_secret_value(**sdk_options)
+
+    def set(
+        self,
+        name: str,
+        value: Union[str, dict, bytes],
+        *,  # force keyword arguments
+        client_request_token: Optional[str] = None,
+        **sdk_options,
+    ) -> SetSecretResponse:
+        """
+        Modify the details of a secret or create a new secret if it doesn't already exist.
+        It includes metadata and the secret value.
+
+        We aim to minimize API calls by assuming that the secret already exists and needs updating.
+        If it doesn't exist, we attempt to create a new one. Refer to the following workflow for a better understanding:
+
+
+                          ┌────────────────────────┐      ┌─────────────────┐
+                ┌───────▶│Resource NotFound error?│────▶│Create Secret API│─────┐
```

Comment:
Can we add a space or two to make the box align? Great explanation btw :D

Reply:
Please take a screenshot of what you especially want. I only partially understand hahaha